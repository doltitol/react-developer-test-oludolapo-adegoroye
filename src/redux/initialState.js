export const initialState = {
  cartItems: JSON.parse(localStorage.getItem('cartItems') || '[]'),
  cartQuantity: JSON.parse(localStorage.getItem('cartItems') || '[]').reduce(
    (acc, item) => acc + item.qty,
    0
  ),
  currency: '$',
};
