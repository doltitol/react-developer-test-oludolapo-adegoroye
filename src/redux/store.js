import { createStore, combineReducers } from 'redux';
import cartReducer from './reducers/cartReducers';

const rootReducer = combineReducers({
  cart: cartReducer,
});
export const store = createStore(rootReducer);
