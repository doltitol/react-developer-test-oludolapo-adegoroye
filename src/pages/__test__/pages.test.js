import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { store } from '../../redux/store';
import Pages from '../';
import React from 'react';
describe('<Pages /> spec', () => {
  it('renders the component', () => {
    const container = render(
      <Provider store={store}>
        <Pages />
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
