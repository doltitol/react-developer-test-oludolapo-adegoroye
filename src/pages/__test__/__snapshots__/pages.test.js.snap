// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pages /> spec renders the component 1`] = `undefined`;
